src/QSymbolicsBase/rules.jl

```diff
@@ -1,18 +1,36 @@
+##
+# This file defines automatic simplification rules for specific operations of quantum objects.
+##
+
+##
+# Predicate functions
+##
 function hasscalings(xs)
     any(xs) do x
         operation(x) == *
     end
 end
+_isa(T) = x->isa(x,T)
+
+##
+# Determining factors for expressions containing quantum objects
+##
 
-""" Used to perform (a*|k⟩) ⊗ (b*|l⟩) → (a*b) * (|k⟩⊗|l⟩) """
-function prefactorscalings(xs)
+function prefactorscalings(xs; scalar=false) # If the scalar keyword is true, then only scalar factors will be returned as coefficients
     terms = []
     coeff = 1::Any
     for x in xs
         if isexpr(x) && operation(x) == *
             c,t = arguments(x)
-            coeff *= c
-            push!(terms,t)
+            if scalar == false
```

Comment:
minor stylistic note, it is a bit more natural to write `if !scalar`. And `if scalar` below.